AppCreationScripts/AppCreationScripts.md

```diff
@@ -0,0 +1,107 @@
+    ```
+4. If you have never done it already, in the PowerShell window, install the AzureAD PowerShell modules. For this, type:
+    ```PowerShell
+    Install-Module AzureAD
+    ```
+5. Go to the `AppCreationScripts` sub-folder. From the folder where you cloned the repo,
+    ```PowerShell
+    cd AppCreationScripts
+    ```
+6. Run the scripts. See below for the [four options](#four-ways-to-run-the-script) to do that.
+7. Open the Visual Studio solution, and in the solution's context menu, choose **Set Startup Projects**.
+8. select **Start** for the projects
+
+You're done. this just works!
+
+### Four ways to run the script
+
+We advise four ways of running the script:
+
+- Interactive: you will be prompted for credentials, and the scripts decide in which tenant to create the objects,
+- non-interactive: you will provide crendentials, and the scripts decide in which tenant to create the objects,
+- Interactive in specific tenant: you will be prompted for credentials, and the scripts decide in which tenant to create the objects,
+- non-interactive in specific tenant: you will provide crendentials, and the scripts decide in which tenant to create the objects.
+
+Here are the details on how to do this.
+
+#### Option 1 (interactive)
+
+- Just run ``. .\Configure.ps1``, and you will be prompted to sign-in (email address, password, and if needed MFA).
+- The script will be run as the signed-in user and will use the tenant in which the user is defined.
+
+Note that the script will choose the tenant in which to create the applications, based on the user. Also to run the `Cleanup.ps1` script, you will need to re-sign-in.
+
+#### Option 2 (non-interactive)
+
+When you know the indentity and credentials of the user in the name of whom you want to create the applications, you can use the non-interactive approach. It's more adapted to DevOps. Here is an example of script you'd want to run in a PowerShell Window
+
+```PowerShell
+$secpasswd = ConvertTo-SecureString "[Password here]" -AsPlainText -Force
+$mycreds = New-Object System.Management.Automation.PSCredential ("[login@tenantName here]", $secpasswd)
+.\Cleanup.ps1 -Credential $mycreds
+.\Configure.ps1 -Credential $mycreds
+```
+
+Of course, in real life, you might already get the password as a `SecureString`. You might also want to get the password from KeyVault.
+
+#### Option 3 (Interactive, but create apps in a specified tenant)
+
+  if you want to create the apps in a particular tenant, you can use the following option:
+- open the [Azure portal](https://portal.azure.com)
+- Select the Azure Active directory you are interested in (in the combo-box below your name on the top right of the browser window)
+- Find the "Active Directory" object in this tenant
+- Go to **Properties** and copy the content of the **Directory Id** property
+- Then use the full syntax to run the scripts:
+
+```PowerShell
+$tenantId = "yourTenantIdGuid"
+.\Cleanup.ps1 -TenantId $tenantId
```

Comment:
I'm really surprised that the . is not required.
which version of PowerShell do you use?

Reply:
Major  Minor  Build  Revision
-----  -----  -----  --------
5      1      16299  251

Reply:
its two (..) instead of (.) dot that causes an issue if someone copies and pastes from the guidance. Hardly a big issue, but I've developed a habit of fixing things as i test along